### ISSUE TITLE
SameSite cookie tests seem to rely on third-party cookies being allowed

### ISSUE THREAD

**Issue by gsnedders** [web-platform-tests/wpt#36913]
see, e.g., [these results on wpt.fyi](https://wpt.fyi/results/cookies/samesite/fetch.https.html?run_id=5107602061524992&run_id=5178955124965376&run_id=5198184062648320&run_id=5159082680385536&run_id=5177490306236416&run_id=5085787184431104)

"Cross-site fetches are cross-site" fails in Safari with: `assert_equals: SameSite=None cookies are always sent. expected (string) "0.17879251053118983" but got (undefined) undefined`

This is the expected behaviour in Safari, and is compliant with [draft-ietf-httpbis-rfc6265bis-11](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-11#section-7.1):

>    While this document does not endorse or require a specific approach,   it is RECOMMENDED that user agents adopt a policy for third-party   cookies that is as restrictive as compatibility constraints permit.   Consequently, resources cannot rely upon third-party cookies being   treated consistently by user agents for the foreseeable future.

**Comment by mikewest** [comment]
I agree that we should probably update these tests to accept a broader range of reasonable results. It would likely be a good idea to incorporate storage access API tests here as well to validate behavior for those browsers that block by default but enable after a prompt.

Do y'all have concrete suggestions for changes?
